src/main/java/io/strimzi/kafka/bridge/http/HttpAdminClientEndpoint.java

```diff
@@ -0,0 +1,155 @@
+import io.strimzi.kafka.bridge.http.model.HttpBridgeError;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.kafka.admin.Config;
+import io.vertx.kafka.admin.ConfigEntry;
+import io.vertx.kafka.admin.TopicDescription;
+import io.vertx.kafka.client.common.ConfigResource;
+import org.apache.kafka.common.errors.UnknownTopicOrPartitionException;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class HttpAdminClientEndpoint extends AdminClientEndpoint {
+
+    private HttpBridgeContext httpBridgeContext;
+
+    public HttpAdminClientEndpoint(Vertx vertx, BridgeConfig bridgeConfig, HttpBridgeContext context) {
+        super(vertx, bridgeConfig);
+        this.httpBridgeContext = context;
+    }
+
+    @Override
+    public void open() {
+        super.open();
+    }
+
+    @Override
+    public void handle(Endpoint<?> endpoint) {
+        this.handle(endpoint, null);
+    }
+
+    @Override
+    public void handle(Endpoint<?> endpoint, Handler<?> handler) {
+        RoutingContext routingContext = (RoutingContext) endpoint.get();
+        switch (this.httpBridgeContext.getOpenApiOperation()) {
+            case LIST_TOPICS:
+                doListTopics(routingContext);
+                break;
+
+            case GET_TOPIC:
+                doGetTopic(routingContext);
+                break;
+
+            default:
+                throw new IllegalArgumentException("Unknown Operation: " + this.httpBridgeContext.getOpenApiOperation());
+
+        }
+    }
+
+    public void doListTopics(RoutingContext routingContext) {
+        listTopics(listTopicsResult -> {
+            if (listTopicsResult.succeeded()) {
+                JsonArray root = new JsonArray();
+                Set<String> topics = listTopicsResult.result();
+                topics.forEach(topic -> {
+                    root.add(topic);
+                });
+                HttpUtils.sendResponse(routingContext, HttpResponseStatus.OK.code(), BridgeContentType.KAFKA_JSON, root.toBuffer());
+            } else {
+                HttpBridgeError error = new HttpBridgeError(
+                        HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                        listTopicsResult.cause().getMessage()
+                );
+                HttpUtils.sendResponse(routingContext, HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                        BridgeContentType.KAFKA_JSON, error.toJson().toBuffer());
+            }
+        });
+    }
+
+    public void doGetTopic(RoutingContext routingContext) {
+        String topicName = routingContext.pathParam("topicname");
+
+        Future<Map<String, TopicDescription>> describeTopicsFuture = Future.future();
```

Comment:
lately, we moved to use `Promise` instead of `Future`. Can you change here as well, please?

Reply:
ok